### DISCUSSION TITLE
Continuous Integration Providers

### DISCUSSION THREAD

**Discussion by airbreather** [NetTopologySuite/NetTopologySuite#463] [General]
Over the past month or so, I've noticed that Travis-CI has been quite slow to start up its builds.  It also only runs on Linux, and one of its builds actually failed for no apparent reason, [build 1138](https://travis-ci.org/github/NetTopologySuite/NetTopologySuite/builds/749066773)... note that this link now goes to a successful build because I had it re-run the exact same build with the exact same commit and it passed.

On the other hand, we've had GitHub Actions running just fine on Linux (and also Windows and Mac OS), and it even lets us associate artifacts with our builds out-of-the-box (whereas Travis-CI only seems to let us upload artifacts to an AWS account, which I don't think we have).

Azure Pipelines and AppVeyor are both perfectly fine CI solutions as well.

Perhaps it's time to stop using Travis-CI as our "main" CI provider?

**Comment by FObermaier** [comment]
We are here on github, so we might as well stick to GitHub Actions.

**Comment by airbreather** [comment]
>> Perhaps it's time to stop using Travis-CI as our "main" CI provider?
>>
> We are here on github, so we might as well stick to GitHub Actions.

OK, then, things to finish up this item (unless we want to change the status quo for any of these, which I don't right now):
- [ ] GitHub Actions should run on every push and on every pull request.
- GitHub Actions should deploy on every successful build of certain special branches.
    - Remember to keep the API key a [secret](https://docs.github.com/en/actions/reference/encrypted-secrets)
    - It should deploy after all 3 environments build successfully, but only from one of them (let's pick `ubuntu-latest` for this).
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
- Fix the Directory.Build.props file.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
- [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml

We should also have all of the above set up consistently for all the v2-series repositories that we care about, which I think are:
- NetTopologySuite
    - [x] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [x] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- NetTopologySuite.IO.PostGis
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- NetTopologySuite.IO.ShapeFile
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- NetTopologySuite.IO.SpatiaLite
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- NetTopologySuite.IO.GeoJSON
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- ProjNet4GeoAPI
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- NetTopologySuite.Features
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- NetTopologySuite.IO.GPX
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- NetTopologySuite.IO.TinyWKB
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- NetTopologySuite.IO.SqlServerBytes
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
- NetTopologySuite.IO.Oracle
    - [ ] GitHub Actions should run on every push and on every pull request.
    - [ ] Deploy to MyGet on every successful build on the `develop` branch.
    - [ ] Deploy to NuGet on every successful build on the `master` branch.
    - [ ] There needs to be a `PropertyGroup` for GitHub Actions like we have for other CI providers.
    - [ ] `NtsOfficialRelease` should be `true` when building the `master` branch on GitHub Actions.
    - [ ] `NtsOfficialRelease` should not be `true` in any other case.
    - [ ] Once everything else is done, disable Travis-CI on the backend and delete .travis.yml
